src/itest/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileE2EITest.java

```diff
@@ -107,19 +133,135 @@ void testAddLink(final String linkType, final String deltaType) throws Exception
+        final Updated updated = actualDocument.getUpdated();
+        final Data companyProfile = actualDocument.getCompanyProfile();
+
+        final String actualLink = filterLinkType(linkType, companyProfile.getLinks());
+
+        assertNull(actualLink);
+        assertNotNull(updated.getAt());
+        assertEquals(deltaType, updated.getType());
+        assertEquals(CONTEXT_ID, updated.getBy());
+        assertEquals(1L, actualDocument.getVersion());
+        assertNotEquals(oldEtag, companyProfile.getEtag());
+        verify(companyProfileApiService).invokeChsKafkaApi(CONTEXT_ID, COMPANY_NUMBER);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "charges",
+            "insolvency",
+            "registers"
+    })
+    void testAddLinkLegacyEndpoint(final String linkType) throws Exception {
+        // given
+        final String expectedLink = String.format("/company/%s/%s", COMPANY_NUMBER, linkType);
+        final String oldEtag = "oldEtag";
+
+        VersionedCompanyProfileDocument document = new VersionedCompanyProfileDocument();
+        document.setId(COMPANY_NUMBER)
+                .setCompanyProfile(new Data()
+                        .etag(oldEtag)
+                        .links(new Links()
+                                .self("/company/" + COMPANY_NUMBER)));
+        document.version(0L);
+
+        companyProfileRepository.insert(document);
+
+        when(companyProfileApiService.invokeChsKafkaApi(any(), any())).thenReturn(new ApiResponse<>(200, null));
+
+        // when
+        final ResultActions result = mockMvc.perform(patch(ADD_LINK_ENDPOINT_LEGACY, COMPANY_NUMBER)
+                .header("ERIC-Identity", "123")
+                .header("ERIC-Identity-Type", "key")
+                .header("ERIC-Authorised-Key-Privileges", "internal-app")
+                .header("x-request-id", CONTEXT_ID)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(makeBaseLegacyLinksCompanyProfile())));
+
+        // then
+        result.andExpect(MockMvcResultMatchers.status().isOk());
+
+        final VersionedCompanyProfileDocument actualDocument = Objects.requireNonNull(mongoTemplate.findById(COMPANY_NUMBER, VersionedCompanyProfileDocument.class));
+        final Updated updated = actualDocument.getUpdated();
+        final Data companyProfile = actualDocument.getCompanyProfile();
+
+        final String actualLink = filterLinkType(linkType, companyProfile.getLinks());
+
+        assertEquals(expectedLink, actualLink);
+        assertNotNull(updated.getAt());
+        assertEquals(CONTEXT_ID, updated.getBy());
+        assertEquals(1L, actualDocument.getVersion());
         assertNotEquals(oldEtag, companyProfile.getEtag());
         verify(companyProfileApiService).invokeChsKafkaApi(CONTEXT_ID, COMPANY_NUMBER);
     }
 
     private String filterLinkType(final String linkType, Links links) {
```

Comment:
This can be refactored to this to remove yellow underlines:

```
    private String filterLinkType(final String linkType, Links links) {
        return switch (linkType) {
            case "charges" -> links.getCharges();
            case "exemptions" -> links.getExemptions();
            case "filing-history" -> links.getFilingHistory();
            case "insolvency" -> links.getInsolvency();
            case "officers" -> links.getOfficers();
            case "persons-with-significant-control" -> links.getPersonsWithSignificantControl();
            case "persons-with-significant-control-statements" -> links.getPersonsWithSignificantControlStatements();
            case "uk-establishments" -> links.getUkEstablishments();
            case "registers" -> links.getRegisters();
            default -> "DID NOT MATCH LINK TYPE";
        };
    }
```